src/SIMD.jl

```diff
@@ -59,19 +59,18 @@ const IntegerTypes = Union{BoolTypes, IntTypes, UIntTypes}
 const FloatingTypes = Union{Float16, Float32, Float64}
 const ScalarTypes = Union{IntegerTypes, FloatingTypes}
 
-typealias VE Base.VecElement
+const VE = Base.VecElement
 
 export Vec
 immutable Vec{N,T<:ScalarTypes} <: DenseArray{T,1}   # <: Number
     elts::NTuple{N,VE{T}}
-    @inline Vec() = new()
-    @inline Vec(elts::NTuple{N,VE{T}}) = new(elts)
+    @inline Vec{N,T}(elts::NTuple{N, VE{T}}) where {N,T} = new(elts)
```

Comment:
How is this syntax backward compatible? I'd expect an `@compat` macro or similar to make things work for 0.5.